dask_kubernetes/operator/tests/test_operator.py

```diff
@@ -0,0 +1,161 @@
+                        "default-worker-group",
+                    )
+                    await client.wait_for_workers(5)
+                    k8s_cluster.kubectl(
+                        "scale",
+                        "--replicas=3",
+                        "daskworkergroup",
+                        "default-worker-group",
+                    )
+                    await client.wait_for_workers(3)
+
+
+@pytest.mark.timeout(180)
+@pytest.mark.asyncio
+async def test_simplecluster(k8s_cluster, kopf_runner, gen_cluster):
+    with kopf_runner as runner:
+        async with gen_cluster() as cluster_name:
+            scheduler_pod_name = "simple-cluster-scheduler"
+            worker_pod_name = "simple-cluster-default-worker-group-worker"
+            while scheduler_pod_name not in k8s_cluster.kubectl("get", "pods"):
+                await asyncio.sleep(0.1)
+            while cluster_name not in k8s_cluster.kubectl("get", "svc"):
+                await asyncio.sleep(0.1)
+            while worker_pod_name not in k8s_cluster.kubectl("get", "pods"):
+                await asyncio.sleep(0.1)
+
+            with k8s_cluster.port_forward(f"service/{cluster_name}", 8786) as port:
+                async with Client(
+                    f"tcp://localhost:{port}", asynchronous=True
+                ) as client:
+                    await client.wait_for_workers(2)
+                    # Ensure that inter-worker communication works well
+                    futures = client.map(lambda x: x + 1, range(10))
+                    total = client.submit(sum, futures)
+                    assert (await total) == sum(map(lambda x: x + 1, range(10)))
+            assert cluster_name
+
+    assert "A DaskCluster has been created" in runner.stdout
+    assert "A scheduler pod has been created" in runner.stdout
+    assert "A worker group has been created" in runner.stdout
+
+
+@pytest.fixture()
+async def gen_cluster2(k8s_cluster):
+    """Yields an instantiated context manager for creating/deleting a simple cluster."""
+
+    @asynccontextmanager
+    async def cm():
+        cluster_path = os.path.join(DIR, "resources", "simplecluster.yaml")
+        cluster_name = "foo"
+        try:
+            yield cluster_name
+        finally:
+            k8s_cluster.kubectl("delete", "-f", cluster_path, "--wait=true")
+            while "foo-cluster" in k8s_cluster.kubectl("get", "daskclusters"):
+                await asyncio.sleep(0.1)
+
+    yield cm
+
+
+from dask_kubernetes.operator.core import KubeCluster2
+
+
+@pytest.fixture
+def cluster(kopf_runner):
+    with kopf_runner as runner:
+        with KubeCluster2(name="foo") as cluster:
+            yield cluster
+
+
+@pytest.fixture
+def client(cluster):
+    with Client(cluster) as client:
+        yield client
+
+
+def test_fixtures_kubecluster2(client, cluster):
+    client.scheduler_info()
+    cluster.scale(1)
+    assert client.submit(lambda x: x + 1, 10).result() == 11
```

Comment:
We add another test that creates a couple of instances of `KubeCluster2` to ensure they don't tread on each other.